src/aleph/http/client.clj

```diff
@@ -473,9 +474,10 @@
 
    Can't use an ApnHandler/ApplicationProtocolNegotiationHandler here,
    because it's tricky to run Manifold code on Netty threads."
-  [{:keys [ssl? remote-address ssl-context ssl-endpoint-id-alg]}]
+  [{:keys [ssl? remote-address ssl-context ssl-endpoint-id-alg proxy-options proxy-connected]}]
   (fn pipeline-builder*
     [^ChannelPipeline pipeline]
+    (add-proxy-handlers pipeline proxy-connected proxy-options ssl?)
```

Comment:
Should be after the SSL handlers, for security, no?

Reply:
As in `.addAfter`? Or code ordering wise? The former won't do the right thing and the latter I'm not sure makes a difference since the pipeline builder is run atomically before anything else happens. Since the proxy handlers are added to the front of the pipeline (via `.addFirst`) and the SSL handler is added to the back, I would tend to leave the code ordering as it is since it mirrors the pipeline order.

Reply:
> Since the proxy handlers are added to the front of the pipeline (via .addFirst)

Dammit. This is one of the most annoying things about netty. I thought we consistentnly used .addLast.

I guess I should say, if we supported encrypted client<->proxy connections, we'd have to ensure there was a second SslHandler before the proxy handlers in the pipeline.

Reply:
> Dammit. This is one of the most annoying things about netty. I thought we consistentnly used .addLast.

Ah, I see! OK AFAICT it doesn't make a difference whether we use `.addFirst` or `.addLast` in this particular case since the proxy handlers are the first thing to be added anyway. I've changed it accordingly to reduce mental overhead for future readers :smile: 

> I guess I should say, if we supported encrypted client<->proxy connections, we'd have to ensure there was a second SslHandler before the proxy handlers in the pipeline.

Indeed! Updated the description of https://github.com/clj-commons/aleph/issues/743 to mention this more explicitly.